### PR TITLE
Implement InputHistory

### LINKED ISSUE #87: 💡 Input history
### Describe the issue
*What is the problem, if any? What is the desired behavior?*
Cycle through input history with up/down button.
Easier for user to retype command if small typo/ if want to type a similar command.
Also easier for us to test.
________________________________________________________________
### Additional context
*Describe possible and/or alternative solutions, to highlight required scope of work.*

### PR DESCRIPTION
**Pull request by garyljj**
closes #87

### CODE REVIEW

File: src/main/java/seedu/hippocampus/commons/util/InputHistory.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    /**
+     * Add new input to history, given it is different from the previous input.
+     * If lst have reached maxsize, remove earliest entry.
+     *
+     * @param s User input.
+     */
+    public void add(String s) {
+        if (lst.isEmpty() || isDifferentFromLastest(s)) {
```
Comment (nickyfoo) [inline src/main/java/seedu/hippocampus/commons/util/InputHistory.java:31]:
```suggestion
        if (lst.isEmpty() || isDifferentFromLatest(s)) {
```

File: src/main/java/seedu/hippocampus/commons/util/InputHistory.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            if (lst.size() == MAXSIZE) {
+                lst.remove(0);
+            }
+            lst.add(s);
+        }
+
+        current = lst.size();
+    }
+
+    private boolean isDifferentFromLastest(String s) {
```
Comment (nickyfoo) [inline src/main/java/seedu/hippocampus/commons/util/InputHistory.java:41]:
```suggestion
    private boolean isDifferentFromLatest(String s) {
```

**Review by nickyfoo — CHANGES_REQUESTED** [review]
Besides the typo, LGTM

We might want to add documentation for this as well, can put up as an issue when we merge this, if we deem it necessary

**Review by pyuxiang — APPROVED** [review]
Easy to review, LGTM. For optimization, can consider using a ring buffer instead of a linked list.

**Review by nickyfoo — APPROVED** [review]

**Pull request merged**